### DISCUSSION TITLE
Design/Styles Definitions Configuration

### DISCUSSION THREAD

**Discussion by PierreFaraut** [zauberzeug/nicegui#1257] [Q&A]
### Question

Hi there,

I'm looking for guidance on how to set up a configuration file effectively for my application. The main objective is to consolidate all design definitions into a single file. To start with, I have created the following Python code, but I'm unsure if it's the best approach:

```python
from nicegui import ui

def app_design():
    ui.query('body').classes('bg-white')
    ui.query('nicegui-content').classes('bg-red')
    
    ui.colors(
        primary='#424242',
        secondary='#26a69a',
        accent='#9c27b0',
        dark='#1d1d1d',
        positive='#21ba45',
        negative='#c10015',
        info='#31ccec',
        warning='#f2c037'
    )

app_design()

ui.run()
```
The current implementation addresses setting primary colors, but I would like to extend it to cover more elements (such as background colors for example). Additionally, I want to include support for dark mode colors in the same manner.

Ultimately, the goal is to separate all design-related aspects from the rest of the application's logic. Do you have any ideas on how to achieve this effectively?

Thank you.

**Comment by rodja** [comment]
Your approach is good, @PierreFaraut. You can also derive your own elements from the NiceGUI defaults to change the appearance. Like @ofenbach is doing it in [CatDesign](https://github.com/ofenbach/CatDesign/blob/main/CatDesign/).
